src/ralph/backends/data/es.py

```diff
@@ -0,0 +1,321 @@
+        logger.error("Cluster status is not green: %s", cluster_status)
+
+        return DataBackendStatus.ERROR
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """List available Elasticsearch indices, data streams and aliases.
+
+        Args:
+            target (str or None): The comma-separated list of data streams, indices,
+                and aliases to limit the request. Supports wildcards (*).
+                If target is `None`, lists all available indices, data streams and
+                    aliases. Equivalent to (`target` = "*").
+            details (bool): Get detailed informations instead of just names.
+            new (bool): Ignored.
+
+        Yields:
+            str: The next index, data stream or alias name. (If `details` is False).
+            dict: The next index, data stream or alias details. (If `details` is True).
+
+        Raises:
+            BackendException: If a failure during indices retrieval occurs.
+        """
+        target = target if target else "*"
+        try:
+            indices = self.client.indices.get(index=target)
+        except (ApiError, ESConnectionError) as error:
+            msg = "Failed to read indices: %s"
+            logger.error(msg, error)
+            raise BackendException(msg % error) from error
+
+        if new:
+            logger.warning("The `new` argument is ignored")
+
+        if details:
+            for index, value in indices.items():
+                yield {index: value}
+
+            return
+
+        for index in indices:
+            yield index
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, ESQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = 500,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Read documents matching the query in the target index and yield them.
+
+        Args:
+            query: (str or ESQuery): A query in the Lucene query string syntax or a
+                dictionary defining a search definition using the Elasticsearch Query
+                DSL. The Lucene query overrides the query DSL if present.
+                See Elasticsearch search reference for Lucene query syntax:
+                https://www.elastic.co/guide/en/elasticsearch/reference/8.6/search-search.html#search-api-query-params-q
+                See Elasticsearch search reference for query DSL syntax:
+                https://www.elastic.co/guide/en/elasticsearch/reference/8.6/search-search.html#request-body-search-query
+            target (str or None): The target Elasticsearch index name to query.
+                If target is `None`, the `default_index` is used instead.
+            chunk_size (int or None): The chunk size for reading batches of documents.
+                If chunk_size is `None` it defaults to 10.
```

Comment:
```suggestion
                If chunk_size is `None` it defaults to 500.
```

Reply:
The default is indeed `500`; however, it seems that when `chunk_size` is set explicitly to `None` the `size` body field of the request to Elasticsearch `_search` path is not added and defaults to `10`: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-size-param

Reply:
However, this is indeed confusing, should we set `chunk_size` to `500` if it is set to `None`?

Reply:
It would be better yes!